src/Symfony/Component/Translation/Translator.php

```diff
@@ -320,7 +316,12 @@ public function getMessages($locale = null)
      */
     protected function loadCatalogue($locale)
     {
-        if (null === $this->cacheDir) {
+        $this->assertValidLocale($locale);
```

Comment:
do we need to assert it here ? We could do the validation only when we don't have the catalogue yet (as currently), thanks to the fact that we will never create ``$this->catalogues[$locale]`` for an invalid locale